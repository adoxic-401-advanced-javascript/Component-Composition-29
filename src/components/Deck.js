import React from 'react';
import Card from './Card';
import PropTypes from 'prop-types';


const Deck = ({ children, title }) => {
  return (
    <section>
      {title &&
      <header>
        <h2>{title}</h2>
      </header>}
      <Card>
        {children}
      </Card>
    </section>
    
  );
};

Deck.propTypes = {
  children: PropTypes.array.isRequired,
  title: PropTypes.string
};

export default Deck;
