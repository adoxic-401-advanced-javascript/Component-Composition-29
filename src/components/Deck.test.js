import React from 'react';
import { shallow } from 'enzyme';
import Deck from './Deck';

describe('Deck component', () => {
  it('renders Deck', () => {
    const item = [{ _id: 'some stuff', name: 'thing' }];
    
    const wrapper = shallow(
      <Deck title='hello'>
        {item}
      </Deck>);
    expect(wrapper).toMatchSnapshot();
  });
});
