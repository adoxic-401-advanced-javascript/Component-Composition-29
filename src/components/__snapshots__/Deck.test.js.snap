// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deck component renders Deck 1`] = `
<section>
  <header>
    <h2>
      hello
    </h2>
  </header>
  <Card>
    <Component />
  </Card>
</section>
`;
